homeassistant/components/shelly/utils.py

```diff
@@ -0,0 +1,15 @@
+"""Shelly helpers functions."""
+
+import logging
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_remove_entity_by_domain(hass, domain, unique_id):
+    """Remove entity by domain."""
+    entity_reg = await hass.helpers.entity_registry.async_get_registry()
+    for entity in entity_reg.entities.values():
```

Comment:
Call this `entry` so it better describes the value
```suggestion
    for entry in entity_reg.entities.values():
```